### PR TITLE
feat: use dropdown sub-nav on desktop for fitness and settings

### PR DESCRIPTION
**Pull request by llun**
## What changed

Desktop **Fitness** and **Settings** pages now use the same **dropdown sub-navigation** that tablet/mobile already used, instead of the side-by-side vertical icon nav rail. The content column now spans the full width on desktop.

- `app/(timeline)/fitness/layout.tsx` and `app/(timeline)/settings/layout.tsx`: removed the `lg:block` vertical rail; a single `<nav>` wrapping a Radix `DropdownMenu` now renders at every breakpoint. The trigger shows the active tab's `text-primary` Lucide icon + label + chevron — `w-full` on mobile, `sm:w-64` from `sm` up.
- `lib/components/page-header.tsx`: refreshed comments that referenced the removed rail.
- Updated both `layout.test.tsx` files to assert the dropdown trigger reflects the active tab.

## Why

Matches the design system (`ui_kits/web`) and the requested behavior: desktop should not have a doubled-up vertical menu beside the main sidebar — the dropdown is used across all breakpoints so the content gets the full frame width.

## Docs

- `AGENTS.md` + `CLAUDE.md`: documented the dropdown pattern and added an explicit rule **not** to reintroduce a vertical nav rail on these desktop sub-pages. Also fixed a dangling reference to the removed `settings/fitness/layout.tsx`.

## Desktop sub-navigation: before → after

Desktop **Fitness** and **Settings** drop the side-by-side vertical icon rail and use the same dropdown sub-nav as tablet/mobile, so the content column spans the full width.

| | Before (vertical rail) | After (dropdown) |
| :-- | :-- | :-- |
| **Fitness** | before-fitness | after-fitness |
| **Settings** | before-settings | after-settings |

_Captured at 1440px width. The dropdown trigger shows the active tab's orange icon + label + chevron; selecting it opens the full list of sections._

## Testing

- `yarn run prettier --write` ✅
- `yarn lint` ✅ (0 errors; 12 pre-existing unrelated warnings)
- `yarn build` ✅
- `yarn test` ✅ (373 suites, 3209 tests)
- Browser-verified at 1440px against local SQLite + mock user: both `/fitness` and `/settings` show the dropdown with the orange active icon, no rail, full-width content; dropdowns open showing all items with icons — matching the attached screenshots.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: app/(timeline)/fitness/layout.tsx [hunk 1]
```diff
@@ -79,45 +82,23 @@ const Layout: FC<Props> = ({ children }) => {
               <DropdownMenuContent className="w-[--radix-dropdown-menu-trigger-width]">
```
[the remark this answers is not in the document]
Comment (llun) [reply app/(timeline)/fitness/layout.tsx]: Done — added align="start" to DropdownMenuContent so the menu aligns with the trigger's left edge.

File: app/(timeline)/fitness/layout.tsx [hunk 2]
```diff
@@ -79,45 +82,23 @@ const Layout: FC<Props> = ({ children }) => {
...
                 {tabs.map((tab) => (
                   <DropdownMenuItem key={tab.url} asChild>
-                    <Link href={tab.url} className="flex items-center gap-2">
+                    <Link
+                      href={tab.url}
+                      aria-current={
+                        tab.url === activeTab.url ? 'page' : undefined
+                      }
+                      className="flex items-center gap-2"
+                    >
```
[the remark this answers is not in the document]
Comment (llun) [reply app/(timeline)/fitness/layout.tsx]: Done — active item now gets font-medium text-primary (design-system orange active state); used cn() instead of string concat.

File: app/(timeline)/settings/layout.tsx [hunk 3]
```diff
@@ -86,48 +89,23 @@ const Layout: FC<Props> = ({ children }) => {
               <DropdownMenuContent className="w-[--radix-dropdown-menu-trigger-width]">
```
[the remark this answers is not in the document]
Comment (llun) [reply app/(timeline)/settings/layout.tsx]: Done — added align="start" to DropdownMenuContent.

File: app/(timeline)/settings/layout.tsx [hunk 4]
```diff
@@ -86,48 +89,23 @@ const Layout: FC<Props> = ({ children }) => {
...
                 {tabs.map((tab) => (
                   <DropdownMenuItem key={tab.url} asChild>
-                    <Link href={tab.url} className="flex items-center gap-2">
+                    <Link
+                      href={tab.url}
+                      aria-current={
+                        tab.url === activeTab.url ? 'page' : undefined
+                      }
+                      className="flex items-center gap-2"
+                    >
```
[the remark this answers is not in the document]
Comment (llun) [reply app/(timeline)/settings/layout.tsx]: Done — active item now gets font-medium text-primary via cn().

File: app/(timeline)/fitness/layout.test.tsx [hunk 5]
```diff
@@ -19,43 +19,27 @@ const renderLayout = () =>
...
-      'aria-current',
-      'page'
-    )
-    expect(
-      within(rail).getByRole('link', { name: 'Overview' })
-    ).not.toHaveAttribute('aria-current')
+    const nav = screen.getByRole('navigation', { name: 'Fitness' })
+    expect(within(nav).getByRole('button')).toHaveTextContent('Privacy')
   })
 })
```
[the remark this answers is not in the document]
Comment (llun) [reply app/(timeline)/fitness/layout.test.tsx:107]: Done — added tests opening the dropdown to assert all menu items render and the active one carries aria-current="page"; added the same to settings. Needed jsdom pointer polyfills + hidden:true queries for Radix.

File: app/(timeline)/settings/layout.test.tsx [hunk 6]
```diff
@@ -19,18 +19,12 @@ const renderLayout = () =>
...
 
-    const rail = screen.getByRole('navigation', { name: 'Settings' })
-    expect(within(rail).getByRole('link', { name: 'Account' })).toHaveAttribute(
-      'aria-current',
-      'page'
-    )
-    expect(
-      within(rail).getByRole('link', { name: 'General' })
-    ).not.toHaveAttribute('aria-current')
+    const nav = screen.getByRole('navigation', { name: 'Settings' })
```
[the remark this answers is not in the document]
Comment (llun) [reply app/(timeline)/settings/layout.test.tsx:26]: Done in cda4e8d — added align="start" to DropdownMenuContent in settings (and fitness).

File: app/(timeline)/fitness/layout.test.tsx [hunk 7]
```diff
@@ -19,43 +19,27 @@ const renderLayout = () =>
...
     ;(usePathname as jest.Mock).mockReturnValue('/fitness')
     renderLayout()
 
-    const rail = screen.getByRole('navigation', { name: 'Fitness' })
-    for (const label of ['Overview', 'Files', 'Privacy', 'Strava']) {
-      expect(
-        within(rail).getByRole('link', { name: label })
-      ).toBeInTheDocument()
-    }
+    const nav = screen.getByRole('navigation', { name: 'Fitness' })
```
[the remark this answers is not in the document]
Comment (llun) [reply app/(timeline)/fitness/layout.test.tsx:26]: Done in cda4e8d — added tests that open the dropdown (via keyboard, matching mobile-nav.test) and assert every section renders as a menu item, plus aria-current on the active one. Restores the coverage the rail had.

File: AGENTS.md [hunk 8]
```diff
@@ -160,20 +161,23 @@ section-navigation patterns; pick by section type.
...
+            <div className="min-w-0">{children}</div>
           </div>
         </PageHeaderSectionProvider>
       </>
     )
   }
   ```
 
-- A **nested** sub-nav inside a rail section (e.g. fitness General/Privacy/Strava) renders as a small **in-content segmented control**, not a second rail. Hand it to the closest section-mode `PageHeader` via `PageSubnavProvider` so it sits directly **below the per-page title** (header-first, like the non-nested pages) rather than above it. See `app/(timeline)/settings/layout.tsx` and `app/(timeline)/settings/fitness/layout.tsx`.
+- A **nested** sub-nav inside a section renders as a small **in-content segmented control**, not a second dropdown or rail. Hand it to the closest section-mode `PageHeader` via `PageSubnavProvider` so it sits directly **below the per-page title** (header-first, like the non-nested pages) rather than above it. See `app/(timeline)/settings/layout.tsx` and `app/(timeline)/fitness/layout.tsx`.
```
[the remark this answers is not in the document]
Comment (llun) [reply AGENTS.md]: Done in cda4e8d — active item now gets font-medium text-primary composed with the cn() utility (matching the rest of the codebase).

[the remark this answers is not in the document]
Comment (llun) [reply AGENTS.md]: Correcting my earlier reply (it was mis-pasted): fixed in the latest commit — AGENTS.md now points the nested PageSubnavProvider example at app/(timeline)/admin/layout.tsx (the actual live user) and notes the settings/fitness layouts use the dropdown instead.

File: app/(timeline)/settings/layout.test.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (llun) [reply app/(timeline)/settings/layout.test.tsx:26]: Correcting my earlier reply (mis-pasted): the aria-current coverage is addressed in cda4e8d — the settings layout test now opens the dropdown and asserts the active item has aria-current="page" (and a non-active one does not).

**Comment by llun** [comment]
@codex review

**Comment by llun** [comment]
## Self-review loop (subagent code review) — converged after 4 rounds

I ran multiple independent subagent reviewers (correctness, design-system consistency, tests, edge-cases, accessibility) over the diff across rounds and addressed every genuine in-scope finding. Summary:

### Round 1 — findings addressed
- **Code duplication (medium):** the dropdown sub-nav was copy-pasted between the settings and fitness layouts. → Extracted a shared `lib/components/section-nav-dropdown.tsx` (`SectionNavDropdown`), both layouts now consume it. Commit `6f9bdf9d8`.
- **Test scoping (medium):** opened-menu assertions used the global `screen` scope. → Scoped to the open `menu` role via `within(menu)` so tests fail loudly if items render outside an open dropdown. Commit `6f9bdf9d8`.
- **Missing fitness nested-path coverage (low):** → Added `/fitness/files/abc123 → Files` test mirroring the settings nested-path test. Commit `6f9bdf9d8`.

### Round 2 — findings addressed
- **No dedicated test for the new component (low):** → Added co-located `lib/components/section-nav-dropdown.test.tsx` (nav label, longest-prefix active resolution, first-tab fallback, all items render, aria-current). Commit `8d9ed7a09`.
- **AGENTS.md didn't mention the shared component (low):** → Updated the "Page Header & Sub-Navigation" section to direct contributors to reuse `SectionNavDropdown` and updated the code sketch. Commit `8d9ed7a09`.

### Round 3 — **NO FINDINGS** (clean)

### Round 4 — only pre-existing / out-of-scope items (no changes made)
- **`/fitness/heatmap` shows "Overview" in the trigger:** `heatmap` has no tab, so the active-tab logic falls back to the first tab. This behavior is **identical to the previous vertical-rail implementation** (not a regression) and whether `/fitness/heatmap` should be a tab is a product decision outside this PR's scope. Flagged for visibility.
- **Two `<nav>` landmarks / sidebar label & `text-primary` contrast (low, verify-only):** both pre-existing — the prior rail already rendered a second `aria-label`'d `<nav>`, and `text-primary` is the design-system's documented active color. Not introduced by this PR.

### Verification at HEAD (`8d9ed7a09`)
- `prettier` ✅ · `yarn lint` ✅ (0 errors) · `yarn build` ✅ · `yarn test` ✅ **(374 suites, 3219 tests)**

Also requested fresh external review via `/gemini review` and `@codex review` above.

**Comment by llun** [comment]
## Review loop — update (converged after an additional round)

My earlier summary called convergence too early. Two later subagent rounds (edge-case + accessibility lenses) surfaced **two genuine issues I then fixed** — correcting the record:

### Additional round — findings addressed (commit `4ad22a3`)
- **Active menu-item contrast (medium, introduced by this PR):** the active item used `text-primary` (orange) on white ≈ 3.3:1, below WCAG AA. → Restored the design system's documented signature active state **`bg-primary/10 text-primary`** (the same orange wash the old rail used). The wash is a non-color-dependent cue alongside `aria-current`, so the state no longer leans on orange text alone.
- **Brittle test trigger query (medium):** the dropdown-open tests used an unscoped `screen.getByRole('button')`. → Scoped to `within(nav).getByRole('button')` so they don't depend on the trigger being the only button on the page.
- **Missing `href` assertions (low):** added `href` checks on menu items (active item + every tab in the component test) so a tab-URL regression can't slip through.
- **Fitness header-position coverage (low):** added a header-above-nav test mirroring the settings one.

### Final round — **NO FINDINGS**

### Out of scope (flagged, not changed)
- `/fitness/heatmap` shows "Overview" in the trigger because `heatmap` has no tab. This is **identical to the previous vertical-rail behavior** (same tab set, confirmed via `git show main`), so it's not a regression; whether Heatmap should be its own tab is a separate product decision.

### Verification at HEAD (`4ad22a3`)
- `prettier` ✅ · `yarn lint` ✅ (0 errors) · `yarn build` ✅ · `yarn test` ✅ **(374 suites, 3221 tests)**
